### DISCUSSION TITLE
Rex: ConnectionRefused

### DISCUSSION THREAD

**Discussion by v3cn4x00** [rapid7/metasploit-framework#18047] [Q&A] [answered]
When I'm trying to run exploit command in meterpreter msf6. The output is giving error exploit failed and unreachable. The connection iwas refused bt the remote host 
![IMG_20230601_011433.jpg](https://github.com/rapid7/metasploit-framework/assets/97059953/08f6423b-4387-475b-a259-5c07ce0ab8a8)

![IMG_20230601_011513.jpg](https://github.com/rapid7/metasploit-framework/assets/97059953/31f81c28-fe4c-43bd-8a3f-96a27af24b02)

**Comment by bcoles** [comment]
The connection to port 21 was refused. Ensure port 21 is open.

**Reply by v3cn4x00** [reply]
How to do that can u guide please

**Reply by bcoles** [reply]
In the Metasploit screenshot you are attempting to connect to 192.168.0.100 on port 21 and 192.168.0.103 on port 21.

In the nmap screenshot you are attempting to connect to 192.168.0.100 on port 445.

**Reply by v3cn4x00** [reply]
Yea so what should I do now use port 445 in metasploit

**Reply by bcoles** [reply]
You can set the remote port to 445 with the `set` command.

```
set rport 445
```

**Reply by v3cn4x00** [reply]
Thanks alot brother

**Reply by v3cn4x00** [reply]
![IMG_20230601_124326.jpg](https://github.com/rapid7/metasploit-framework/assets/97059953/dc19b12c-726a-4bd8-888d-a6087085e9ba)

What to do in this situation bro

**Reply by bcoles** [reply]
> What to do in this situation bro

I have no idea what you are trying to do. I don't think you do either. Figure out what you are trying to do first.

**Reply by v3cn4x00** [reply]
I'm using metasploit to gain access of my device. I'm using command sudo msfconsole to run metasploit. After that I'm using search vsftpd 
![IMG_20230601_131834.jpg](https://github.com/rapid7/metasploit-framework/assets/97059953/8448835e-fb58-4cd8-a1da-fe40e8b15709)


Then I'm using ( use exploit/Unix/ftp/vsftpd_234_backdoor)
![IMG_20230601_132108.jpg](https://github.com/rapid7/metasploit-framework/assets/97059953/efe759c5-66ef-424b-abae-6fc5c6d2704f)


Here I'm setting RPORT as 445 as you mentioned
Then setting RHOSTS to target host IP

I did nmap scan to search for open ports
This i done before setting RHOSTS and RPORT
![IMG_20230601_132419.jpg](https://github.com/rapid7/metasploit-framework/assets/97059953/986a85d4-21a2-4f91-82a3-abe6938bde33)

Setting payload cmd/unix/interact 
Then exploit
 I changed the RPORT to 62078
Then run exploit again
![IMG_20230601_133002.jpg](https://github.com/rapid7/metasploit-framework/assets/97059953/0698961e-ab4d-40b9-b906-449999718715)

**Comment by bcoles** [comment]
> Then I'm using ( use exploit/Unix/ftp/vsftpd_234_backdoor)

Why? Are you running a very old version of vsftpd which contains the backdoor?

> Here I'm setting RPORT as 445 as you mentioned

Why? FTP servers usually listen on port 21 rather than 445.

> This i done before setting RHOSTS and RPORT

The nmap output shows no ports are open on 192.168.0.100.

> I changed the RPORT to 62078

Why? Port 62078 is not open on 192.168.0.100.

**Reply by v3cn4x00** [reply]
No I just installed the new version of metasploit

**Reply by v3cn4x00** [reply]
But port 21 was not responding I showed u also

**Reply by v3cn4x00** [reply]
It was open now at time of photo it was close

**Reply by v3cn4x00** [reply]
![IMG_20230601_142237.jpg](https://github.com/rapid7/metasploit-framework/assets/97059953/2a4b52ba-8e71-4958-a07e-5db20228cdbc)


See this and sorry for inconvenience

**Reply by bcoles** [reply]
What makes you think that any of the hosts on the network are running a backdoored version of vsftp?

None of your screenshots show port 21 open.

None of your screenshots show vsftpd.

**Reply by v3cn4x00** [reply]
Should I provide u with a link. You can see that and guide me please 

https://youtu.be/QynUOJanNqo

See this from 6.40 please

**Reply by v3cn4x00** [reply]
![IMG_20230601_144007.jpg](https://github.com/rapid7/metasploit-framework/assets/97059953/ef69aa6f-f686-4ebe-88cd-bd1a13838522)

**Reply by v3cn4x00** [reply]
And can you please tell me whether we can open port for certain target or not

Very very sorry if your time is wasting 🙏🙏

**Reply by bcoles** [reply]
> Should I provide u with a link. You can see that and guide me please
> 
> https://youtu.be/QynUOJanNqo
> 
> See this from 6.40 please

I presume you are trying to gain a Metasploit session on Metasploitable2 using the vsftpd_234_backdoor module.

I suggest re-watching the video from 7 minutes to 8 minutes.

**Reply by v3cn4x00** [reply]
Yea you are right ok I will watch again

**Reply by v3cn4x00** [reply]
![IMG_20230604_001043.jpg](https://github.com/rapid7/metasploit-framework/assets/97059953/37c42785-c553-4b84-a897-492eb0002fa5)

![IMG_20230604_000950.jpg](https://github.com/rapid7/metasploit-framework/assets/97059953/37d4d4d6-b1bf-431b-b633-0f4462dc8d2f)


Brother check these. By the way I watched the video as you mentioned

**Reply by bcoles** [reply]
Modules targeting TCP network services must be able to establish a TCP connection.

If a TCP module cannot establish a TCP connection then the module cannot communicate with the network service using TCP.

If the module cannot communicate with the network service using TCP, the module cannot send and receive data to and from the network service.

If the module cannot send and receive data to and from the network service, the module cannot send exploit code.

If the module cannot send exploit code, the module cannot exploit the service.

Those screenshots show port `21` is `filtered` on `192.168.0.100`. The port must appears as `open`.

**Reply by v3cn4x00** [reply]
Ok thanks alot for explaining

**Comment by bcoles** [comment, marked as answer]
`Rex: ConnectionRefused` means the connection was refused. Ensure you are connecting to the correct host and port (`rhosts` and `rport`).
